dlt/destinations/impl/dremio/sql_client.py

```diff
@@ -0,0 +1,154 @@
+        raise NotImplementedError("You cannot rollback Dremio SQL statements.")
+
+    @property
+    def native_connection(self) -> "pydremio.DremioConnection":
+        return self._conn
+
+    def drop_tables(self, *tables: str) -> None:
+        # Tables are drop with `IF EXISTS`, but dremio raises when the schema doesn't exist.
+        # Multi statement exec is safe and the error can be ignored since all tables are in the same schema.
+        with suppress(DatabaseUndefinedRelation):
+            super().drop_tables(*tables)
+
+    def execute_sql(
+        self, sql: AnyStr, *args: Any, **kwargs: Any
+    ) -> Optional[Sequence[Sequence[Any]]]:
+        with self.execute_query(sql, *args, **kwargs) as curr:
+            if curr.description is None:
+                return None
+            else:
+                f = curr.fetchall()
+                return f
+
+    @contextmanager
+    @raise_database_error
+    def execute_query(self, query: AnyStr, *args: Any, **kwargs: Any) -> Iterator[DBApiCursor]:
+        db_args = args if args else kwargs if kwargs else None
+        with self._conn.cursor() as curr:
+            try:
+                curr.execute(query, db_args)  # type: ignore
+            except pydremio.MalformedQueryError as ex:
+                raise DatabaseTransientException(ex)
+            yield DremioCursorImpl(curr)  # type: ignore
+
+    def fully_qualified_dataset_name(self, escape: bool = True) -> str:
+        database_name = self.credentials.database
+        dataset_name = self.dataset_name
+        if escape:
+            database_name = self.capabilities.escape_identifier(database_name)
+            dataset_name = self.capabilities.escape_identifier(dataset_name)
+        return f"{database_name}.{dataset_name}"
+
+    def make_qualified_table_name(self, table_name: str, escape: bool = True) -> str:
+        if escape:
+            table_name = self.capabilities.escape_identifier(table_name)
+        return f"{self.fully_qualified_dataset_name(escape=escape)}.{table_name}"
+
+    @classmethod
+    def _make_database_exception(cls, ex: Exception) -> Exception:
+        if isinstance(ex, pyarrow.lib.ArrowInvalid):
+            msg = str(ex)
+            if "not found in any table" in msg:
+                return DatabaseTerminalException(ex)
+            elif "not found" in msg or "does not exist" in msg:
+                return DatabaseUndefinedRelation(ex)
+            elif "Non-query expression encountered in illegal context" in msg:
+                return DatabaseTransientException(ex)
+            else:
+                return DatabaseTerminalException(ex)
+        else:
+            return ex
+
+    @staticmethod
+    def is_dbapi_exception(ex: Exception) -> bool:
+        return isinstance(ex, (pyarrow.lib.ArrowInvalid, pydremio.MalformedQueryError))
+
+    def create_dataset(self) -> None:
+        table_name = self.make_qualified_table_name("_dlt_dataset_exists")
```

Comment:
why exactly do we need this table to be created? You probably mentioned it already.. in any case it needs to go into the docs that this happens.

Reply:
I've removed this after some further consideration. This was necessary to get the `test_default_schema_name_init_storage` test to pass. It relates back to the lack of `CREATE SCHEMA`. The tests are being run using a "NAS" data source which allows us to circumvent the `CREATE SCHEMA` issue, as it can create tables at arbitrary paths in the file system. I was using this dummy table as a marker that dataset had been created. I think this is probably unnecessary, and I've simply disabled the `test_default_schema_name_init_storage` test for Dremio.